### PR TITLE
refactor(chip)!: remove deprecated properties and values.

### PR DESCRIPTION
**Pull request by anveshmekala**
BREAKING CHANGE: Removed the `dismissible` property and the `clear` value for the `appearance` property.

- Removed the property `dismissible`, use `closable` instead.
- Use the value `transparent` instead of `clear` for `appearance` property.

### CODE REVIEW

**Comment by anveshmekala** [comment]
> @anveshmekala can you check for any readme, html, story, and test files that need to be updated?
> 
> Can you change the description/title to:
> 
> ```
> BREAKING CHANGE: Removed the `dismissible` property.
> 
> - Removed the property `dismissible`, use `closable` instead.
> ```

For readme, we can rely on update-doc Github action.

**Comment by driskull** [comment]
> For readme, we can rely on update-doc Github action.

@anveshmekala can you update the usage file markdown?

File: src/components/chip/usage/Basic.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 ```html
-<calcite-chip value="Global" dismissible icon="globe" appearance="clear" color="green">Global</calcite-chip>
+<calcite-chip value="Global" closable icon="globe" appearance="clear" color="green">Global</calcite-chip>
```
Comment (driskull) [inline src/components/chip/usage/Basic.md:2]: @anveshmekala one more thing...

I think the `appearance` value of `clear` is deprecated. Can we change them all to `transparent`?

Comment (driskull) [reply src/components/chip/usage/Basic.md:2]: We could always do this later if you would like that better.

Comment (anveshmekala) [reply src/components/chip/usage/Basic.md:2]: Good catch. will change it to `transparent`. And get the `conditionalSlotComponent` removed as well.

Comment (driskull) [reply src/components/chip/usage/Basic.md:2]: We can do the `connectConditionalSlotComponent` in a separate pr/epic. Lets wait on that one.

**Pull request merged**